javalib/src/main/scala/java/util/Arrays.scala

```diff
@@ -150,260 +258,159 @@ object Arrays {
   }
 
   @noinline def binarySearch(a: Array[Short], key: Short): Int =
-    binarySearchImpl[Short](a, 0, a.length, key, _ < _)
+    binarySearchImpl[Short](a, 0, a.length, key)
 
   @noinline def binarySearch(a: Array[Short], startIndex: Int, endIndex: Int, key: Short): Int = {
     checkRangeIndices(a, startIndex, endIndex)
-    binarySearchImpl[Short](a, startIndex, endIndex, key, _ < _)
+    binarySearchImpl[Short](a, startIndex, endIndex, key)
   }
 
   @noinline def binarySearch(a: Array[Char], key: Char): Int =
-    binarySearchImpl[Char](a, 0, a.length, key, _ < _)
+    binarySearchImpl[Char](a, 0, a.length, key)
 
   @noinline def binarySearch(a: Array[Char], startIndex: Int, endIndex: Int, key: Char): Int = {
     checkRangeIndices(a, startIndex, endIndex)
-    binarySearchImpl[Char](a, startIndex, endIndex, key, _ < _)
+    binarySearchImpl[Char](a, startIndex, endIndex, key)
   }
 
   @noinline def binarySearch(a: Array[Byte], key: Byte): Int =
-    binarySearchImpl[Byte](a, 0, a.length, key, _ < _)
+    binarySearchImpl[Byte](a, 0, a.length, key)
 
   @noinline def binarySearch(a: Array[Byte], startIndex: Int, endIndex: Int, key: Byte): Int = {
     checkRangeIndices(a, startIndex, endIndex)
-    binarySearchImpl[Byte](a, startIndex, endIndex, key, _ < _)
+    binarySearchImpl[Byte](a, startIndex, endIndex, key)
   }
 
   @noinline def binarySearch(a: Array[Double], key: Double): Int =
-    binarySearchImpl[Double](a, 0, a.length, key, _ < _)
+    binarySearchImpl[Double](a, 0, a.length, key)
 
   @noinline def binarySearch(a: Array[Double], startIndex: Int, endIndex: Int, key: Double): Int = {
     checkRangeIndices(a, startIndex, endIndex)
-    binarySearchImpl[Double](a, startIndex, endIndex, key, _ < _)
+    binarySearchImpl[Double](a, startIndex, endIndex, key)
   }
 
   @noinline def binarySearch(a: Array[Float], key: Float): Int =
-    binarySearchImpl[Float](a, 0, a.length, key, _ < _)
+    binarySearchImpl[Float](a, 0, a.length, key)
 
   @noinline def binarySearch(a: Array[Float], startIndex: Int, endIndex: Int, key: Float): Int = {
     checkRangeIndices(a, startIndex, endIndex)
-    binarySearchImpl[Float](a, startIndex, endIndex, key, _ < _)
+    binarySearchImpl[Float](a, startIndex, endIndex, key)
   }
 
   @noinline def binarySearch(a: Array[AnyRef], key: AnyRef): Int =
-    binarySearchImplRef(a, 0, a.length, key)
+    binarySearchImpl(a, 0, a.length, key)
 
   @noinline def binarySearch(a: Array[AnyRef], startIndex: Int, endIndex: Int, key: AnyRef): Int = {
     checkRangeIndices(a, startIndex, endIndex)
-    binarySearchImplRef(a, startIndex, endIndex, key)
+    binarySearchImpl(a, startIndex, endIndex, key)
   }
 
-  @noinline def binarySearch[T](a: Array[T], key: T, c: Comparator[_ >: T]): Int =
-    binarySearchImpl[T](a, 0, a.length, key, (a, b) => c.compare(a, b) < 0)
+  @noinline def binarySearch[T <: AnyRef](a: Array[T], key: T, c: Comparator[_ >: T]): Int =
```

Comment:
Hum, yes, I think it was wrong before. It's supposed to be this method:
https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#binarySearch-T:A-int-int-T-java.util.Comparator-
but in Java, generic arrays are always object arrays, since generic types must be class types.

There's probably a missing test for this method; otherwise the test suite shouldn't have passed previously.

Reply:
I added new tests in the first commit. They were indeed not linking previous to this change.